src/D2L.CodeStyle.TestAnalyzers/NUnit/CategoryAnalyzer.cs

```diff
@@ -69,9 +70,16 @@ internal sealed class CategoryAnalyzer : DiagnosticAnalyzer {
 					return;
 				}
 
+				var syntaxReference = attribute.ApplicationSyntaxReference;
+
+				if( syntaxReference == null ) {
+					// ??
+					return;
+				}

```

Comment:
Since this already would have been throwing an NRE in this case anyways, I think raising an exception or forcing non-null would be fine here. This property is undocumented, so can't really tell when it would ever actually _be_ null. But better to loudly fail if those cases ever come up than silently ignore and potentially let stuff slip though.